Author: Daniel Richard G. <skunk@iSKUNK.ORG>
Description: Avoid bindgen+libclang crash due to C++23 mode

On jammy, bindgen breaks inside libclang if the -std=c++23 option is
passed, with an error message like the following:

  FAILED: gen/base/logging_log_severity_bindgen_generator/bindings.rs
  [...]
  thread 'main' panicked at 'libclang error; possible causes include:
  - Invalid flag syntax
  - Unrecognized flags
  - Invalid flag arguments
  - File I/O errors
  - Host vs. target architecture mismatch
  If you encounter an error missing from this list, please file an issue or a PR!', src/ir/context.rs:538:15

Reducing the option to -std=c++20 avoids the problem.

Also address an error due to a different option:

  error: '-ftrivial-auto-var-init=zero' hasn't been enabled; enable it at your own peril for benchmarking purpose only with '-enable-trivial-auto-var-init-zero-knowing-it-will-be-removed-from-clang', err: true

--- a/build/rust/gni_impl/run_bindgen.py
+++ b/build/rust/gni_impl/run_bindgen.py
@@ -66,6 +66,10 @@ def main():
       nargs="*")
   args = parser.parse_args()
 
+  # XtraDeb workaround to avoid libclang breakage
+  args.clangargs = ['-std=c++20' if arg == '-std=c++23' else arg \
+      for arg in args.clangargs if arg != '-ftrivial-auto-var-init=zero']
+
   # Abort if `TARGET` exists in the environment. Cargo sets `TARGET` when
   # running build scripts and bindgen will try to be helpful by using that value
   # if it's set. In practice we've seen a case where someone had the value set
